### PR TITLE
Invite to battle

### PR DESCRIPTION
**Pull request by pantoja**
## Description

- Created a function to send an email to the opponent containing a battle invitation
- Refactored the change battle status code (moved from forms to views).
- Added tests for sent email and change battle status functions. 

## Trello Card

As a selected opponent I receive an email when I'm invited to a match
## Screenshots (if appropriate):
image

## Steps to reproduce (if appropriate):
URL: https://staging-pantoja-pokebattle.herokuapp.com/
(using staging because heroku branches crashes with email functions)
1. Sign up with an email that you have access to to check the inbox
2. Sign out, sign in with another account or email: `admin@admin.com` password: `senhaadmin`
3. Create a battle and the opponent must be the user you signed up with in step 1
4. Proceed to create a team successfully. The email will be sent after creating a team.
5. Checkout your inbox, it should have an email with an invite to reply the battle.

### CODE REVIEW

File: backend/battles/tests/test_views.py [hunk 1]
```diff
@@ -189,27 +189,72 @@ def test_user_not_invited_to_battle_is_redirected(self):
...
+            "battle": self.battle.id,
+            "pokemon_1": mommy.make("pokemon.Pokemon", name="ivysaur").id,
+            "pokemon_2": mommy.make("pokemon.Pokemon", name="bulbasaur").id,
+            "pokemon_3": mommy.make("pokemon.Pokemon", name="pikachu").id,
+            "order_1": "1",
+            "order_2": "2",
+            "order_3": "3",
+        }
+        self.client.post(self.view_url, post_data)
+        self.assertEqual(len(mail.outbox), 1)
```
Comment (rsarai) [inline backend/battles/tests/test_views.py:217]: nice!

File: backend/battles/views.py [hunk 2]
```diff
@@ -43,6 +50,25 @@ def get_context_data(self, **kwargs):
...
+        if self.object.trainer == battle.user_creator:
+            send_invite_to_match(
+                creator.email,
+                opponent.email,
+                self.request.build_absolute_uri(f"/create-team/{battle.id}"),
+            )
+        if self.object.trainer == battle.user_opponent:
+            result = run_battle(creator.teams.get(battle=battle.pk), self.object)
+            change_battle_status(battle, result["winner"].trainer)
+            send_result_email(result, self.request.build_absolute_uri("/"))
```
Comment (rsarai) [inline backend/battles/views.py:68]: nice! didn't know `build_absolute_uri`

File: backend/battles/views.py [hunk 3]
```diff
@@ -43,6 +50,25 @@ def get_context_data(self, **kwargs):
 
         return context
 
+    def form_valid(self, form):
+        self.object = form.save()
+        battle = self.object.battle
+        creator = battle.user_creator
+        opponent = battle.user_opponent
+
+        if self.object.trainer == battle.user_creator:
```
Comment (rsarai) [inline backend/battles/views.py:59]: Thinking if here is the right place for this. Shouldn't you be using celery to send these emails?

Comment (pantoja) [reply backend/battles/views.py:59]: Yeah actually that change is made in the next trello card that asks to run battle asynchronously. [The changes were made here](https://github.com/pantoja/PokeBattle/blob/async-battle/backend/battles/views.py#L65) if you want to check out.

File: backend/battles/views.py (hunk 2 above)
Comment (pantoja) [reply backend/battles/views.py:68]: It was Amanda that told me about it!

File: backend/battles/views.py (hunk 3 above)
Comment (rsarai) [reply backend/battles/views.py:59]: ok!

**Pull request merged**